includes/admin/settings/register-settings.php

```diff
@@ -996,11 +996,6 @@ function edd_settings_sanitize_misc_accounting( $input ) {
 		return $input;
 	}
 
-	if( edd_get_file_download_method() != $input['download_method'] || ! edd_htaccess_exists() ) {
```

Comment:
It does fire, in the proper section https://github.com/easydigitaldownloads/Easy-Digital-Downloads/blob/master/includes/admin/settings/register-settings.php#L967